hypertrace-alert-engine/src/integrationTest/java/org/hypertrace/alert/engine/HypertraceAlertEngineTest.java

```diff
@@ -165,43 +167,95 @@ public static void setup() throws Exception {
-        .execute(
-            () ->
-                IntegrationTestServerUtil.startServices(new String[] {"hypertrace-alert-engine"}));
-    Assertions.assertEquals(0, mockWebServer.getRequestCount());
   }
 
   @AfterAll
-  public static void shutdown() throws IOException {
+  public static void shutdown() {
     LOG.info("Initiating shutdown");
     attributeService.stop();
     queryService.stop();
     mongo.stop();
     pinotServiceManager.stop();
     kafkaZk.stop();
     network.close();
-    mockWebServer.close();
   }
 
   @Test
-  public void testNotificationSent() throws InterruptedException {
+  public void testStaticThresholdNotificationSent() throws Exception {
+    LOG.info("Starting testStaticThresholdNotificationSent");
+    MockWebServer mockWebServer = new MockWebServer();
+    mockWebServer.start(NOTIFICATION_CHANNEL_PORT1);
+    MockResponse mockedResponse =
+        new MockResponse().setResponseCode(200).addHeader("Content-Type", "application/json");
+    mockWebServer.enqueue(mockedResponse);
+
+    Assertions.assertEquals(0, mockWebServer.getRequestCount());
+    withEnvironmentVariable("ATTRIBUTE_SERVICE_HOST_CONFIG", attributeService.getHost())
+        .and("ATTRIBUTE_SERVICE_PORT_CONFIG", attributeService.getMappedPort(9012).toString())
+        .and("QUERY_SERVICE_HOST_CONFIG", queryService.getHost())
+        .and("QUERY_SERVICE_PORT_CONFIG", queryService.getMappedPort(8090).toString())
+        .and("ALERT_RULES_PATH", "build/resources/integrationTest/alert-rules-static-rule.json")
+        .and(
+            "NOTIFICATION_CHANNELS_PATH",
+            "build/resources/integrationTest/notification-channels-1.json")
+        .and("SERVICE_ADMIN_PORT", "10005")
+        .and("JOB_SUFFIX", "job1")
+        .execute(
+            () -> {
+              ConfigFactory.invalidateCaches();
+              IntegrationTestServerUtil.startServices(new String[] {"hypertrace-alert-engine"});
+            });
     // int x = 1;
     int retryCount = 0;
-    while (!(mockWebServer.getRequestCount() > 0) && retryCount++ < 10) {
+    while (!(mockWebServer.getRequestCount() == 1) && retryCount++ < 10) {
       Thread.sleep(Duration.ofSeconds(20).toMillis());
     }
-    LOG.info("Test completed. No of retries {}", retryCount);
     Assertions.assertTrue(retryCount < 10);
+    String notificationBody1 =
+        mockWebServer.takeRequest().getBody().readString(Charset.defaultCharset());
+    LOG.info(
+        "Static Threshold Test, No of retries {}, staticThreshold notification {}",
+        retryCount,
+        notificationBody1);
+
+    Assertions.assertTrue(notificationBody1.contains("static threshold"));
+    mockWebServer.close();
+  }
+
+  @Test
+  public void testDynamicThresholdNotificationSent() throws Exception {
+    LOG.info("Starting testDynamicThresholdNotificationSent");
+    MockWebServer mockWebServer = new MockWebServer();
+    mockWebServer.start(NOTIFICATION_CHANNEL_PORT2);
```

Comment:
Did you observe an issue using the same port - `NOTIFICATION_CHANNEL_PORT` - as the previous test closes `mockWebServer`?

Reply:
no, the way I intend to assert requires me to do so